packages/firestore/src/api/database.ts

```diff
@@ -341,9 +338,7 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
     validateExactNumberOfArgs('Firestore.settings', arguments, 1);
     validateArgType('Firestore.settings', 'object', 1, settingsLiteral);
 
-    if (
-      objUtils.contains(settingsLiteral as objUtils.Dict<{}>, 'persistence')
-    ) {
+    if (settingsLiteral.hasOwnProperty('persistence')) {
```

Comment:
If the user has overridden `hasOwnProperty` this validation result cannot be trusted. Also, if the object was created with `Object.create(null)` the object has no prototype and this invocation will throw. Calling `Object.prototype.hasOwnProperty.call` is the safer way to phrase this. Since we do this in a few places and the reference can't be minified, it's probably cheaper to keep the `contains` method.

Since this is really validation-related rather than general object manipulation it could make sense to move `contains` into `input_validation.ts`.